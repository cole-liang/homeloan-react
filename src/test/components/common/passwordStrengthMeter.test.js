import React from "react";
import { shallow } from "enzyme";
import PasswordStrengthMeter from "../../../components/common/passwordStrengthMeter";

describe("PasswordStrengthMeter Component Test", () => {
  const weakPassword = "liang1";
  const fairPassword = "liang1col";
  const goodPassword = "liang1cole";
  const strongPassword = "liang1cole123";

  it("should render PasswordStrengthMeter with Weak label", () => {
    const props = { password: weakPassword };
    const wrapper = shallow(<PasswordStrengthMeter {...props} />);
    expect(
      wrapper
        .find("Bootstrap(ProgressBar)")
        .at(0)
        .prop("label")
    ).toBe("Weak");
    expect(wrapper).toMatchSnapshot();
  });

  it("should render PasswordStrengthMeter with Fair label", () => {
    const props = { password: fairPassword };
    const wrapper = shallow(<PasswordStrengthMeter {...props} />);
    expect(
      wrapper
        .find("Bootstrap(ProgressBar)")
        .at(0)
        .prop("label")
    ).toBe("Fair");
    expect(wrapper).toMatchSnapshot();
  });

  it("should render PasswordStrengthMeter with Good label", () => {
    const props = { password: goodPassword };
    const wrapper = shallow(<PasswordStrengthMeter {...props} />);
    expect(
      wrapper
        .find("Bootstrap(ProgressBar)")
        .at(0)
        .prop("label")
    ).toBe("Good");
    expect(wrapper).toMatchSnapshot();
  });

  it("should render PasswordStrengthMeter with Strong label", () => {
    const props = { password: strongPassword };
    const wrapper = shallow(<PasswordStrengthMeter {...props} />);
    expect(
      wrapper
        .find("Bootstrap(ProgressBar)")
        .at(0)
        .prop("label")
    ).toBe("Strong");
    expect(wrapper).toMatchSnapshot();
  });
});
