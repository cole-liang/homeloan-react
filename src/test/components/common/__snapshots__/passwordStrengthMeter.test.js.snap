// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordStrengthMeter Component Test should render PasswordStrengthMeter with Fair label 1`] = `
<Bootstrap(ProgressBar)
  label="Fair"
  now={50}
  variant="warning"
/>
`;

exports[`PasswordStrengthMeter Component Test should render PasswordStrengthMeter with Good label 1`] = `
<Bootstrap(ProgressBar)
  label="Good"
  now={75}
  variant="info"
/>
`;

exports[`PasswordStrengthMeter Component Test should render PasswordStrengthMeter with Strong label 1`] = `
<Bootstrap(ProgressBar)
  label="Strong"
  now={100}
  variant="success"
/>
`;

exports[`PasswordStrengthMeter Component Test should render PasswordStrengthMeter with Weak label 1`] = `
<Bootstrap(ProgressBar)
  label="Weak"
  now={25}
  variant="danger"
/>
`;
